src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSObject.References.cs

```diff
@@ -168,16 +133,15 @@ private void DisposeThis()
 
         ~JSObject()
         {
-            DisposeThis();
+            DisposeImpl(false);
```

Comment:
should these both be false? shouldn't one of the DisposeImpl calls be true?

Reply:
meaning is: "should I skip cleanup of the JS side". 
This is true only when resolving/rejecting promise and caller and callee made the cleanup each on his side.